### PR TITLE
Netstat and Netdev decoders

### PR DESCRIPTION
**Pull request by sathieu**

### CODE REVIEW

**Comment by sathieu** [comment]
I'm still not sure about the field names. Comments welcome.

File: sandbox/lua/decoders/linux_netdev.lua [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    Fields = nil
+}
+
+function process_message()
+    local payload = read_message("Payload")
+
+    local data = grammar:match(payload)
+
+    if not data then
+        return -1
```
Comment (rafrombrc) [inline sandbox/lua/decoders/linux_netdev.lua]: You can return a string error message as the second return value here, would be nice for users to know that the input failed to match the grammar.

File: sandbox/lua/decoders/linux_netdev.lua [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+local grammar = l.Ct(
+                header
+              * l.Ct(line^1)
+              )
+
+local field_names = nil
+
+local msg = {
+    Timestamp = nil,
+    Type = "stats.netstat",
```
Comment (rafrombrc) [inline sandbox/lua/decoders/linux_netdev.lua]: Should this be `stats.netdev` to distinguish it from the messages generated by the other decoder?

File: sandbox/lua/decoders/linux_netstat.lua [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    Fields = nil
+}
+
+function process_message()
+    local payload = read_message("Payload")
+
+    local data = grammar:match(payload)
+
+    if not data then
+        return -1
```
Comment (rafrombrc) [inline sandbox/lua/decoders/linux_netstat.lua]: Again, it'd be great to return an error message here.

**Comment by rafrombrc** [comment]
The code implementing the core functionality here looks quite good to me. I've tested both decoders and they seem to be working well. Thanks for your valuable contribution!

Other than the small notes I made in-line to the code, there are some non-code related updates needed before this is ready for merging:

- Need an entry in the `features` section for v0.10.0 in the CHANGES.txt changelog.
- Need to add a stub file for each decoder that pulls the doc comments into the HTML and man page documentation in the `docs/source/config/decoders` folder. You can look at one of the other sandbox decoder pages in there for guidance.
- Need to add an item to the `index.rst` and `index_noref.rst` files in the same `docs/source/config/decoders` folder above, and also an entry in the `docs/source/sandbox/decoder.rst` file (sorry for the duplication).

Thanks!

**Comment by sathieu** [comment]
@rafrombrc Done. Thanks for your quick review.

NB: I've not tested doc build yet, as my machine is really slow.

**Comment by sathieu** [comment]
doc OK

**Comment by nathwill** [comment]
@sathieu i was just looking for this today, you rock :metal:, thanks!

**Pull request merged**